model/src/main/kotlin/licenses/LicenseView.kt

```diff
@@ -130,7 +130,14 @@ class LicenseView(vararg licenseSources: Set<LicenseSource>) {
 
         return resolvedLicenses.filter { it.license in remainingLicenses }.let { result ->
             if (filterSources) {
-                result.map { it.copy(sources = remainingSources.getValue(it.license)) }
+                result.map { resolvedLicense ->
+                    val remainingOriginalExpressions =
+                        resolvedLicense.originalExpressions.filterKeys {
+                            it in remainingSources.getValue(resolvedLicense.license)
+                        }
```

Comment:
Formatting nit: Could be

                    val remainingOriginalExpressions = resolvedLicense.originalExpressions.filterKeys {
                        it in remainingSources.getValue(resolvedLicense.license)
                    }